### ISSUE TITLE
Blazor CSP - Directive unsafe-inline for styles

### ISSUE THREAD

**Issue by BarkovskiyMaxim** [dotnet/aspnetcore#48204]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

Hello! 
In my application, i use dynamic styles to set the dimensions of the area and have the ability to resize it. I use style='width: @WidthValue, height: @HeightValue' for this purpose. As a result, we get 'unsafe-inline' for styles. 
For example, this functionality is necessary for the ProgressBar component.

### Expected Behavior

There is an opinion that calculated values of styles should be safe.
We have reviewed the code and found that you are using the following construction to apply styles

~~~
this.tryApplySpecialProperty(e, n, s, r) || n.setAttribute(s, o.attributeValue(r))
~~~

For example you can change this code to this: 

~~~
this.tryApplySpecialProperty(e, n, s, r) || (() => {
    if(s === 'style')
        n.style.cssText = o.attributeValue(r)
    else 
        n.setAttribute(s, o.attributeValue(r))
})(); 
~~~

or you can add new attribute like 'safe-style' and check this into this code.

**Comment by javiercn** [comment]
@BarkovskiyMaxim thanks for contacting us.

If I understood correctly, you are putting text in the `@style` attribute within an element, and that is causing issues with CSP? That seems to be a problem within your app.

Blazor does not special case the style attribute in any way, and the proper way to do this would be to set the properties via JS interop, for example.

**Comment by BarkovskiyMaxim** [comment]
@javiercn Yes, that's correct, but if we consider Blazor as a tool for building web applications with the reactive principles, it looks strange. If we look at JS frameworks, as far as I know, some of them provide mechanisms to apply styles safely. I understand that this solution can be implemented through JSInterop, but in that case, I would be managing the reactive application of styles on the client myself. In my opinion, in this context, having the ability to set style=@SomeProperty doesn't make sense.

**Comment by javiercn** [comment]
@BarkovskiyMaxim I don't believe any JS framework supports this directly.

What typically happens is that there is a compile step that transforms the things that you put inside the `style` tag into some JS that gets emitted.

**Comment by BarkovskiyMaxim** [comment]
@javiercn  Yes, that's correct if we talk about attribute. 

For example, in the case of Knockout, when we specify data-bind='style: { width: $data.width }', it will be executed safely because Knockout itself works with its own data-bind attributes. If I remember correctly, the same story applies to React, as the value setting is handled by internal mechanisms. Essentially, a similar scenario can be implemented in Blazor because the style is not directly applied but through internal mechanisms (blazor.server.js). In any case, it can be done on our side using JSInterop, but I believe it would be convenient to have it out-of-the-box.
